### PR TITLE
Added -o flag to output to file

### LINKED ISSUE #208: Add -o option to set the output file
and -o - to output on stdout

### PR DESCRIPTION
**Pull request by byi8220**
Fixes #208.

Passes `cargo test` on multiple passes.

A notable change is that in this implementation the functions in `output.rs` no longer lock stdout for the entire duration from the start, instead writing to stdout normally without an explicit lock. I believe this is fine because the function that is called is the only thing that actually writes to output.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -64,6 +64,8 @@ fn main() {
     let mut path_mapping_file = "";
     let mut filter_option = None;
     let mut num_threads = num_cpus::get() * 2;
+    let mut output_file_path = "-";
```
Comment (marco-c) [inline src/main.rs]: I'd make this an option, None by default and in output.rs then you can use stdio when this is None.

Comment (byi8220) [reply src/main.rs]: I made it an option (None by default) in the newest commit.

File: src/main.rs [hunk 2]
```diff
@@ -191,6 +194,16 @@ fn main() {
                 .parse()
                 .expect("Number of threads should be a number");
             i += 1;
+        } else if args[i] == "-o"{
+            if args.len() <= i + 1{
+                eprintln!("[ERROR]: Output file not specified.\n");
+                print_usage(&args[0]);
+                process::exit(1);
+            }
+            if &args[i + 1] != "-"{
```
Comment (marco-c) [inline src/main.rs]: You can now remove this check

File: src/main.rs [hunk 3]
```diff
@@ -64,6 +65,8 @@ fn main() {
     let mut path_mapping_file = "";
     let mut filter_option = None;
     let mut num_threads = num_cpus::get() * 2;
+    let mut output_file_path: Option<&str> = None;
```
Comment (marco-c) [inline src/main.rs]: Nit: no need to specify the type, this way you can also avoid the `use Option`

File: src/main.rs (hunk 2 above)
Comment (byi8220) [reply src/main.rs]: So to confirm, I shouldn't check for the special case "-o -" to write to stdout?

**Pull request merged**